narwhals/_pandas_like/dataframe.py

```diff
@@ -698,21 +698,38 @@ def to_numpy(self, dtype: Any = None, copy: bool | None = None) -> Any:
             # pandas default differs from Polars, but cuDF default is True
             copy = self._implementation is Implementation.CUDF
 
+        to_convert = set()
+        for key, val in self.schema.items():
+            if val == self._dtypes.Datetime and val.time_zone is not None:  # type: ignore[attr-defined]
+                to_convert.add(key)
+        if to_convert:
+            df = self.with_columns(
+                *[
+                    self.__narwhals_namespace__()
+                    .col(key)
+                    .dt.convert_time_zone("UTC")
+                    .dt.replace_time_zone(None)
+                    for key in to_convert
+                ]
+            )._native_frame
```

Comment:
Is the loop even needed? `col` should handle lists directly:

```suggestion
            df = self.with_columns(
                    self.__narwhals_namespace__()
                    .col(list(to_convert))
                    .dt.convert_time_zone("UTC")
                    .dt.replace_time_zone(None)
            )._native_frame
```